apis/workloads/v1alpha1/instanceset_types.go

```diff
@@ -158,6 +178,20 @@ type InstanceSetSpec struct {
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 
+	// Specifies the desired Ordinals of each InstanceTemplate.
+	// The Ordinals used to specify the ordinal of the instance (pod) names to be generated under an InstanceTemplate.
+	//
+	// For example, if Ordinals are {ranges: [{start: 0, end: 1}], discrete: [7]}, then the instance names generated under this InstanceTemplate would be
+	// $(cluster.name)-$(component.name)-$(template.name)-0、$(cluster.name)-$(component.name)-$(template.name)-1 and
+	// $(cluster.name)-$(component.name)-$(template.name)-7
+	//
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=name
+	TemplatesOrdinals []InstanceTemplateOrdinals `json:"templatesOrdinals,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name"`
```

Comment:
The API should be structured as this:
```
type InstanceSetSpec struct {

    ...

    Ordinals Ordinals `json:"ordinals,omitempty"`

    ...

}


type InstanceTemplate {

    ...

    Ordinals Ordinals `json:"ordinals,omitempty"`
    ...
}
```
Otherwise, the default template (with an empty string name) would be exposed to API users :
```
templatesOrdinals:
   - "" : 
        ranges:
           start:  1
           end: 3
```

This is relatively complex to understand.